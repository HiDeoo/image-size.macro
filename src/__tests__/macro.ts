import plugin from 'babel-plugin-macros'
import pluginTester from 'babel-plugin-tester'

/**
 * The macro path & import.
 */
const macroPath = "'../../lib/index.macro'"
const macroImport = `import imageSize from ${macroPath};`
const fixtureDirectoryPath = './__fixtures__'
const imageFixturePath = `'${fixtureDirectoryPath}/image.png'`

describe('validateReferences', () => {
  const error = 'image-size.macro must be used with a default import.'

  pluginTester({
    plugin,
    babelOptions: { filename: __filename },
    tests: {
      'should throw if a default import is not used': {
        code: `
          import { thing } from ${macroPath};
          const size = thing();
        `,
        error,
      },
      'should throw if a named import is used with a default import': {
        code: `
          import imageSize, { thing } from ${macroPath};
          const size = imageSize();
        `,
        error,
      },
      'should throw if only a named import is used': {
        code: `
          import { thing } from ${macroPath};
          const size = thing();
        `,
        error,
      },
    },
  })
})

describe('validateReference', () => {
  pluginTester({
    plugin,
    babelOptions: { filename: __filename },
    tests: {
      'should throw if the macro is not used as a function call': {
        code: `
          ${macroImport}
          const size = imageSize;
        `,
        error: 'image-size.macro must be used as a function call.',
      },
    },
  })
})

describe('getImageSize', () => {
  pluginTester({
    plugin,
    babelOptions: { filename: __filename },
    snapshot: true,
    tests: {
      'should throw if the image does not exist': {
        code: `
          ${macroImport}
          const size = imageSize('${fixtureDirectoryPath}/random.png');
        `,
        error: 'The image path provided to image-size.macro does not exist.',
        snapshot: false,
      },
      'should throw if the file is not a valid image': {
        code: `
          ${macroImport}
          const size = imageSize('${fixtureDirectoryPath}/invalid.png');
        `,
        error: /image-size\.macro could not measure image size: /,
        snapshot: false,
      },
      'should return the image size for a valid file': {
        code: `
          ${macroImport}
          const size = imageSize(${imageFixturePath});
        `,
      },
    },
  })
})
