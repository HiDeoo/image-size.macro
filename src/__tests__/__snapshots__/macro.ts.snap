// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getImageSize macros should return the image size for a valid file: should return the image size for a valid file 1`] = `

import imageSize from '../../lib/index.macro';
const size = imageSize('./__fixtures__/image.png');

      ↓ ↓ ↓ ↓ ↓ ↓

const size = {
  height: 4,
  width: 2,
}


`;
